src/context/context.tsx

```diff
@@ -0,0 +1,108 @@
+"use client";
+import { Dispatch, SetStateAction, createContext, useState } from "react";
+
+import { ProductDetails, STATUS } from "@/constants";
+
+type CategoryStatus = {
+  category: string;
+  status: string;
+};
+
+type CategoryContextType = {
+  categoryStatus: CategoryStatus | null;
+  setCategoryStatus: Dispatch<SetStateAction<CategoryStatus | null>>;
+};
+
+export const CategoryStatusContext = createContext<CategoryContextType>({
+  categoryStatus: null,
+  setCategoryStatus: () => {
+    console.log("in context status", status);
+  },
+});
+
+export const CategoryStatusProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [categoryStatus, setCategoryStatus] = useState<CategoryStatus | null>(
+    null,
+  );
+
+  return (
+    <CategoryStatusContext.Provider
+      value={{ categoryStatus, setCategoryStatus }}
+    >
+      {children}
+    </CategoryStatusContext.Provider>
+  );
+};
+
+type SubCategoryStatus = {
+  name: string;
+  status: STATUS;
+};
+
+type SubCategoryContextType = {
+  subCategoryStatus: SubCategoryStatus | null;
+  setSubCategoryStatus: Dispatch<SetStateAction<SubCategoryStatus | null>>;
+};
+
+export const ActionButtonContext = createContext<SubCategoryContextType>({
+  subCategoryStatus: { name: "", status: STATUS.NONE },
+  setSubCategoryStatus: (status) => {
+    console.log("in context status", status);
+  },
+});
+
+export const SubCategoryStatusProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [subCategoryStatus, setSubCategoryStatus] =
+    useState<SubCategoryStatus | null>(null);
+
+  return (
+    <ActionButtonContext.Provider
+      value={{ subCategoryStatus, setSubCategoryStatus }}
+    >
+      {children}
+    </ActionButtonContext.Provider>
+  );
+};
+
+type PantryProduct = {
+  key: {
+    key: string;
+    product: ProductDetails[];
+    status: string;
+  };
+  value: ProductDetails;
+};
```

Comment:
Thanks for setting this to a typescript file. I was struggling with importing the ProductDetails type for PantryProductProvider.